SparkPost/2web2ui
# FE-1369 - Add custom useSparkPostMutation hook

[FE-1369](https://sparkpost.atlassian.net/browse/FE-1369)

### What Changed

- Adds supporting Cypress tests for the Suppressions create page
- Adds an index file for typeahead components for easier exporting
- Adds global `useAlert` hook
- Adds `useSparkPostMutation` hook
- Refactors the Suppressions create form found on `/lists/suppressions/create` using functional components, `react-hook-form`, and `react-query` with the new `useSparkPostMutation` hook

### How To Test

- Go to `/lists/suppressions/create` and verify the page works as expected when creating a new suppression
- Verify that the new Suppressions create specs adequately test the creation forms
- Verify the `useSparkPostMutation` tests adequately test the new hook
- Read the docs for `useSparkPostMutation` - do you know how to use it?
- Is it clear how to use the new `useAlert` hook?

### To Do

- [X] Refactor Suppression create page to use the new `useSparkPostMutation` hook
- [x] Add supporting Jest tests for `useSparkPostMutation`
